src/loadgenerator/locustfile.py

```diff
@@ -14,115 +14,59 @@
-                "country": "United States",
-                "credit_card_number": "4432-8015-6152-0454",
-                "credit_card_expiration_month": "1",
-                "credit_card_expiration_year": "2039",
-                "credit_card_cvv": "672",
-             }
-    mcCC = {
-                "email": "someone@example.com",
-                "street_address": "1600 Amphitheatre Parkway",
-                "zip_code": "94043",
-                "city": "Mountain View",
-                "state": "CA",
-                "country": "United States",
-                "credit_card_number": "5328897010174228",
-                "credit_card_expiration_month": "1",
-                "credit_card_expiration_year": "2039",
-                "credit_card_cvv": "123",
-        
-    }
-    badCC = {
-                "email": "amex@example.com",
-                "street_address": "1600 Amphitheatre Parkway",
-                "zip_code": "94043",
-                "city": "Mountain View",
-                "state": "CA",
-                "country": "United States",
-                "credit_card_number": "347572753801901",
-                "credit_card_expiration_month": "1",
-                "credit_card_expiration_year": "2026",
-                "credit_card_cvv": "528",
-             }
-    goodCards = [visaCC, mcCC, visaCC, mcCC, visaCC, mcCC, badCC, badCC]
-    goodWithBadCards = [visaCC, mcCC, badCC, badCC, badCC, badCC, badCC]
-    
-    addToCart(l)
-
-    if (counter <= 200):
-        l.client.post(route, random.choice(goodCards))
-    else:
-        l.client.post(route, random.choice(goodWithBadCards))
-    counter += 1
-    if counter >= 300:
-        counter = 0
-
-class UserBehavior(TaskSet):
     def on_start(self):
-        index(self)
-
-    tasks = {
-        index: 5,
-        browseProduct: 5,
-        addToCart: 1,
-        viewCart: 1,
-        checkout: 2,
-    }
-
-
-class WebsiteUser(HttpLocust):
-    task_set = UserBehavior
-    min_wait = 1000
-    max_wait = 10000
+        self.index()
+    
+    @task(50)
+    def index(self):
+        self.client.get('/')
+
+    @task(50)
+    def browse_product(self):
+        self.client.get("/product/" + random_product())
+
+    @task(10)
+    def add_to_cart(self):
+        product = random_product()
+        self.client.get("/product/" + product)
+        self.client.post(
+            "/cart", {"product_id": product, "quantity": random_quantity()}
+        )
+
+    @task(10)
+    def view_cart(self):
+        self.client.get("/cart")
+
+    @task(20)
+    def checkout(self):
+        self.add_to_cart()
+        self.client.post("/cart/checkout", random_card(bad=random.random() < 0.01))
+
+    @task(1)
+    def set_currency(self):
+        self.client.post("/setCurrency", {"currency_code": random_currency()})
+
+
+class HourlyWave(LoadTestShape):
+    fast_low_end = 10
+    fast_high_end = 30
+
+    slow_low_end = 0
+    slow_high_end = 50
+
+    def tick(self):
```

Comment:
Sorry possibly dumb question - what exactly is the "tick" doing each time, and will the "tick" be called every hour? Or how were you thinking this would work exactly?

Reply:
(Just checked in the locust code and this is called every second, except while the number of users is being ramped up or down to match the new value.)